.github/workflows/check.yml

```diff
@@ -1,71 +1,66 @@
-name: "Check CML GitHub Action"
-
+name: Check CML GitHub Action
 on: [push]
-
 jobs:
   check:
-    if: "!contains(github.event.head_commit.message, '[ci skip]')"
+    defaults:
+      run:
+        shell: bash
     strategy:
       matrix:
-        runs-on: [ '{"runs-on": "ubuntu-latest"}', 
-          '{"runs-on": "macos-latest"}', 
-          '{"runs-on": "ubuntu-latest", "container": "ubuntu:18.04"}', 
-          '{"runs-on": "windows-latest"}']
-
-    runs-on: ${{ fromJSON(matrix.runs-on).runs-on }}
-    container: ${{ fromJSON(matrix.runs-on).container }} 
-
+        system:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        include:
+          - system: ubuntu-18.04
+            container: ubuntu:18.04
+    runs-on: ${{ matrix.system }}
+    container: ${{ matrix.container }}
     steps:
-      - name: "git"
-        if: "fromJSON(matrix.runs-on).container == 'ubuntu:18.04'"
+      - name: install deps
+        if: matrix.container == 'ubuntu:18.04'
         run: |
           apt update -y
-          apt install software-properties-common -y 
+          apt install -y software-properties-common
           add-apt-repository ppa:git-core/ppa
-          apt -y update && apt install -y git
-
+          apt update -y
+          apt install -y git fontconfig make gcc pkg-config
```

Comment:
should we document these deps?

Reply:
#23; probably (?)

Reply:
alternatively we could add this to the action itself?

Reply:
They're there just because the Ubuntu Docker images are really bare-bones and doesn't include them. We can include them in the action, but that would imply including operating system checks. What if users choose a container with Alpine Linux? 🤔

Reply:
It would be safer to either ignore or document them.